### PR TITLE
fix: update verify process

### PR DESCRIPTION
**Pull request by cyLi-Tiger**
I think the verification process needs another condition. According to the prove in [SpecInfer](https://arxiv.org/pdf/2305.09781.pdf), when all candidates in one draft candidate path are accepted, we need to sample the `sample_token` from the `logits` of the last accepted node. Here, you still sample from the `gtp` and the last accepted token is also sampled from `gtp`, which may cause repetition in the output. Here is what I got from your example.

image

Please correct me if I misunderstood anything!

### CODE REVIEW

**Comment by Liyuhui-12** [comment]
Thank you.
